Sources/Classes/RSDeviceModeManager.m

```diff
@@ -37,14 +46,14 @@ - (void) startDeviceModeProcessor:(NSArray<RSServerDestination*>*) destinations
     [RSLogger logDebug:@"RSDeviceModeManager: DeviceModeProcessor: Initializing the Custom Factories"];
     [self initiateCustomFactories];
     [RSLogger logDebug:@"RSDeviceModeManager: DeviceModeProcessor: Replaying the message queue to the Factories"];
-    [self replayMessageQueue];
     self->areFactoriesInitialized = YES;
     // initaiting the transformation processor only if there are any factories passed have a device mode transformation connected to them on control plane
     if([self doFactoriesPassedHaveTransformationsEnabled]){
         RSDeviceModeTransformationManager* deviceModeTransformationManager = [[RSDeviceModeTransformationManager alloc] initWithConfig:self->config andDBPersistentManager:self->dbPersistentManager andDeviceModeManager:self andNetworkManager:self->networkManager];
         [RSLogger logDebug:@"RSDeviceModeManager: DeviceModeProcessor: Starting the Device Mode Transformation Processor"];
         [deviceModeTransformationManager startTransformationProcessor];
     } else {
+        [self replayMessageQueue];
```

Comment:
Why are we doing the replay only when `doFactoriesPassedHaveTransformationsEnabled` is false ?

Reply:
When `doFactoriesPassedHaveTransformationsEnabled` is `true` then DMT will take care of events, right? It fetches the events from the DB and send it to the respective destinations after transformation.
Do we really need to replayMessageQueue in that case?

Reply:
We need replayMessaeQueue in both the cases because not every destination has transformation, and if any destination is not having transformations then the DMT response will not include the events for that destination and those events will never be sent to destinations with transformations

Reply:
Got that, I'll move back the `replayMessaeQueue` to its previous place.